### PR TITLE
Add write_to_model function for model scripts

### PR DESCRIPTION
**Pull request by chamini2**
`write_to_model` function is partially applied before being sent to the model script, that way the user cannot write to the wrong model.

TODO
- [x]  add tests

### CODE REVIEW

File: src/faldbt/project.py [hunk 1]
```diff
@@ -358,6 +359,23 @@ def _find_features(self) -> List[Feature]:
                 )
         return features
 
+    def _model(
```
Comment (turbo1912) [inline src/faldbt/project.py:362]: should we call these `_get_model` and the other one `_get_source` non very method looks weird

**Review by turbo1912 — APPROVED** [review]
super excited for this

File: src/faldbt/project.py (hunk 1 above)
Comment (chamini2) [reply src/faldbt/project.py:362]: we can do that for sure. I have a refactor PR coming later, we can do it here

File: src/fal/fal_script.py [hunk 2]
```diff
@@ -26,21 +27,28 @@ def exec(self, context, faldbt: FalDbt):
...
+                "list_models": faldbt.list_models,
+                "list_models_ids": faldbt.list_models_ids,
+                "list_sources": faldbt.list_sources,
+                "list_features": faldbt.list_features,
+                "el": faldbt.el,
+            }
+
+            if self.model is not None:
+                # Hard-wire the model
+                exec_globals["write_to_model"] = partial(
```
Comment (mkam039) [inline src/fal/fal_script.py:45]: TIL `partial`. This is great!

**Review by mkam039 — APPROVED** [review]

**Pull request merged**